Userland/Applications/PixelPaint/Tools/MoveTool.cpp

```diff
@@ -49,37 +49,75 @@ void MoveTool::on_mousemove(Layer* layer, MouseEvent& event)
     if (!layer)
         return;
 
-    constexpr int sensitivity = 20;
-    auto bottom_right_layer_coordinates = layer->relative_rect().bottom_right().translated(1);
-    auto bottom_right_frame_coordinates = m_editor->content_to_frame_position(bottom_right_layer_coordinates).to_rounded<int>();
-    auto grab_rect_top_left = bottom_right_frame_coordinates.translated(-sensitivity / 2);
-    auto grab_rect = Gfx::IntRect(grab_rect_top_left, Gfx::IntSize(sensitivity, sensitivity));
-    auto updated_is_in_lower_right_corner = grab_rect.contains(event.raw_event().position()); // check if the mouse is in the lower right corner
-    if (m_mouse_in_resize_corner != updated_is_in_lower_right_corner) {
-        m_mouse_in_resize_corner = updated_is_in_lower_right_corner;
-        m_editor->update_tool_cursor();
+    if (!m_scaling) {
+        auto updated_is_in_lower_right_corner = mouse_in_resize_corner(layer, event);
+        if (m_mouse_in_resize_corner != updated_is_in_lower_right_corner) {
+            m_mouse_in_resize_corner = updated_is_in_lower_right_corner;
+            m_editor->update_tool_cursor();
+        }
     }
 
     if (!m_layer_being_moved)
         return;
 
+    auto cursor_position = event.image_event().position();
+    auto delta = cursor_position - m_event_origin;
+    int x;
```

Comment:
You could use an IntRect for these variables x, y, width, and height. Perhaps you could assign directly to the m_new_layer_rect ?

Reply:
updated to assign directly to `m_new_layer_rect`